### PR TITLE
docs: modify gke deployment

### PR DESCRIPTION
**Pull request by jacky-xbb**

### CODE REVIEW

File: docs/en_US/deployment/on-gcp-gke.md [hunk 1]
```diff
@@ -1,60 +1,160 @@
...
 - [Private GKE Cluster](https://cert-manager.io/docs/installation/compatibility/#gke)
 
-Don't forget to install CRDs when running `helm` using `--set installCRDs=true`.
+Remember to install CRDs when running `helm` with the `--set installCRDs=true` flag.
 
-> More info at [cert-manager](https://cert-manager.io).
+> More information can be found at [cert-manager](https://cert-manager.io).
 
-## EMQX Operator
+To install `emqx-operator`, refer to the official [docs](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/getting-started/getting-started.md)
```
Comment (Rory-Z) [inline docs/en_US/deployment/on-gcp-gke.md:28]: To install `emqx-operator`, refer to the official [docs](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/getting-started/getting-started.md)

::: tip

The EMQX Operator relies on `cert-manager`,  to install `cert-manager`, consult the official documentation:

- [GKE Autopilot](https://cert-manager.io/docs/installation/compatibility/#gke-autopilot)
- [Private GKE Cluster](https://cert-manager.io/docs/installation/compatibility/#gke)

Remember to install CRDs when running `helm` with the `--set installCRDs=true` flag.
:::

File: docs/en_US/deployment/on-gcp-gke.md [hunk 2]
```diff
@@ -1,60 +1,160 @@
...
-## Create EMQX Cluster
+```Shell
+kubectl get sc
+```
+We use `standard` as storage class in the guide
 
-[Operator installation reference](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/getting-started/getting-started.md)
 
-After Operator is installed, deploy EMQX cluster in GKE as usual.
+### Deploying EMQX Cluster
```
Comment (Rory-Z) [inline docs/en_US/deployment/on-gcp-gke.md:39]: ## Enable EMQX Cluster Persistence

File: docs/en_US/deployment/on-gcp-gke.md [hunk 3]
```diff
@@ -1,60 +1,160 @@
...
+apiVersion: apps.emqx.io/v2alpha1
+kind: EMQX
+metadata:
+  name: emqx
+spec:
+  image: "emqx:5.0"
+  coreTemplate:
+    spec:
+      volumeClaimTemplates:
+        storageClassName: standard
```
Comment (Rory-Z) [inline docs/en_US/deployment/on-gcp-gke.md:55]: The default storageCalssName in Google is `standard` ?

File: docs/en_US/deployment/on-gcp-gke.md [hunk 4]
```diff
@@ -1,60 +1,160 @@
...
+
+
+### Veirty the deployment
+
+:::: tabs type:card
+::: tab v2alpha1
+
+- Retrieve the listener IP address of the load balancer:
+```Shell
+kubectl get svc emqx-listeners -o json | jq '.status.loadBalancer.ingress'
```
Comment (Rory-Z) [inline docs/en_US/deployment/on-gcp-gke.md:111]:
```suggestion
kubectl get svc emqx-listeners -o json | jq '.status.loadBalancer.ingress[0].IP'
```

File: docs/en_US/deployment/on-gcp-gke.md [hunk 5]
```diff
@@ -1,60 +1,160 @@
...
 ```
+
+- Retrieve the dashboard IP address of the load balancer:
+```Shell
+kubectl get svc emqx-listeners -o json | jq '.status.loadBalancer.ingress'
```
Comment (Rory-Z) [inline docs/en_US/deployment/on-gcp-gke.md:116]:
```suggestion
kubectl get svc emqx-dashboard -o json | jq '.status.loadBalancer.ingress[0].IP'`

File: docs/en_US/deployment/on-gcp-gke.md (hunk 3 above)
Comment (jacky-xbb) [reply docs/en_US/deployment/on-gcp-gke.md:55]: yes

File: docs/en_US/deployment/on-gcp-gke.md (hunk 2 above)
Comment (jacky-xbb) [reply docs/en_US/deployment/on-gcp-gke.md]: `Get an available storage class` is better

File: docs/en_US/deployment/on-gcp-gke.md [hunk 6]
```diff
@@ -1,60 +1,174 @@
...
 
-## Configure Cert Manager
+## Deploy EMQX on GKE
 
-To install `cert-manager` read the official docs:
+### Deploying EMQX Operator
+
+To install EMQX Operator, please check [Quick Start](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/getting-started/getting-started.md)
+
+Deploying Cert Manager
```
Comment (Rory-Z) [inline docs/en_US/deployment/on-gcp-gke.md:19]: Delete this line

File: docs/en_US/deployment/on-gcp-gke.md [hunk 7]
```diff
@@ -1,60 +1,174 @@
...
 
 - [GKE Autopilot](https://cert-manager.io/docs/installation/compatibility/#gke-autopilot)
 - [Private GKE Cluster](https://cert-manager.io/docs/installation/compatibility/#gke)
 
-Don't forget to install CRDs when running `helm` using `--set installCRDs=true`.
+Remember to install CRDs when running `helm` with the `--set installCRDs=true` flag.
+:::
 
-> More info at [cert-manager](https://cert-manager.io).
+> More information can be found at [cert-manager](https://cert-manager.io).
```
Comment (Rory-Z) [inline docs/en_US/deployment/on-gcp-gke.md:30]: I think this line should in tip part

**Review by Rory-Z — COMMENTED** [review]
I think next we have to add more LoadBalance content on google cloud

File: docs/en_US/deployment/on-gcp-gke.md [hunk 8]
```diff
@@ -1,60 +1,172 @@
-# Deploy EMQX on Google Cloud GKE
+# Deploy EMQX on Google Kubernetes Engine
 
-Google Kubernetes Engine (GKE) provides a managed environment for deploying, managing, and scaling your containerized applications using Google infrastructure. The GKE environment consists of multiple machines (specifically, Compute Engine instances) grouped together to form a cluster.
+## Overview
 
-## Create GKE Cluster
+This guide will walk you through the process of deploying EMQ X, an open-source MQTT broker, on Google Kubernetes Engine (GKE). By following these steps, you'll learn how to create and configure an EMQ X deployment on GKE.
```
Comment (LenaLenaPan) [inline docs/en_US/deployment/on-gcp-gke.md:5]:
```suggestion
This guide will walk you through the process of deploying EMQ X, an open-source MQTT broker, on Google Kubernetes Engine (GKE). By following these steps, you'll learn how to create and configure an EMQX deployment on GKE.
```

File: docs/en_US/deployment/on-gcp-gke.md [hunk 9]
```diff
@@ -1,60 +1,172 @@
...
++ A Google Kubernetes Engine (GKE) cluster, for more information, please check [Creating an autopilot cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/creating-an-autopilot-cluster)
++ MQTTX CLI, A user-friendly MQTT 5.0 command line tool, download it [here](https://mqttx.app/cli)
 
-For details: [kubeconfig](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl)
 
-## Configure Cert Manager
+## Deploy EMQX on GKE
 
-To install `cert-manager` read the official docs:
+### Deploying EMQX Operator
```
Comment (LenaLenaPan) [inline docs/en_US/deployment/on-gcp-gke.md:15]:
```suggestion
### Deploy EMQX Operator
```

File: docs/en_US/deployment/on-gcp-gke.md [hunk 10]
```diff
@@ -1,60 +1,172 @@
...
+
+To install EMQX Operator, please check [Quick Start](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/getting-started/getting-started.md)
+
+::: warning
```
Comment (LenaLenaPan) [inline docs/en_US/deployment/on-gcp-gke.md:19]:
```suggestion
::: tip
```

File: docs/en_US/deployment/on-gcp-gke.md [hunk 11]
```diff
@@ -1,60 +1,172 @@
...
+To install `cert-manager`, consult the official documentation:
```
Comment (LenaLenaPan) [inline docs/en_US/deployment/on-gcp-gke.md:20]:
```suggestion
For the detailed steps on installing `cert-manager`, see the official documentation:
```

File: docs/en_US/deployment/on-gcp-gke.md [hunk 12]
```diff
@@ -1,60 +1,172 @@
...
 
-> More info at [cert-manager](https://cert-manager.io).
+More information can be found at [cert-manager](https://cert-manager.io).
+:::
 
-## EMQX Operator
+### Enable EMQX Cluster Persistence
 
-Deploy it as usual.
+Google Kubernetes Engine (GKE) now supports a new storage class designed specifically for EMQX message broker deployments. This GKE storage class enhances high-availability and performance, ensuring seamless MQTT message handling and efficient resource utilization in IoT and edge computing environments.
```
Comment (LenaLenaPan) [inline docs/en_US/deployment/on-gcp-gke.md:32]: should we move this to the beginning of the page?

File: docs/en_US/deployment/on-gcp-gke.md [hunk 13]
```diff
@@ -1,60 +1,172 @@
...
+```
+Outputs:
+```Shell
+NAME                        PROVISIONER                    RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
+standard                    kubernetes.io/gce-pd           Delete          Immediate              true                   47h
+...
+```
 
-## Create EMQX Cluster
+Here, we pick up `standard` as storage class in the guide
```
Comment (LenaLenaPan) [inline docs/en_US/deployment/on-gcp-gke.md:43]:
```suggestion
This guide uses the `standard` storage class.
```

File: docs/en_US/deployment/on-gcp-gke.md [hunk 14]
```diff
@@ -1,60 +1,172 @@
...
 
-[Operator installation reference](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/getting-started/getting-started.md)
+For more storage classes, please check [gcp docs](https://cloud.google.com/kubernetes-engine/docs/concepts/persistent-volumes#storageclasses)
 
-After Operator is installed, deploy EMQX cluster in GKE as usual.
 
-### Troubleshoot: Context deadline exceeded when calling "mutating.apps.emqx.io" webhook
+### Deploying EMQX Cluster
```
Comment (LenaLenaPan) [inline docs/en_US/deployment/on-gcp-gke.md:48]:
```suggestion
### Deploy EMQX Cluster
```

File: docs/en_US/deployment/on-gcp-gke.md [hunk 15]
```diff
@@ -1,60 +1,172 @@
...
+  serviceTemplate:
+    spec:
+      type: LoadBalancer
+```
+
+:::
+::::
+
+
+### Veirty the deployment
```
Comment (LenaLenaPan) [inline docs/en_US/deployment/on-gcp-gke.md:114]:
```suggestion
### Verity the Deployment
```

File: docs/en_US/deployment/on-gcp-gke.md [hunk 16]
```diff
@@ -1,60 +1,172 @@
...
+```Shell
+lb_listener_ip=$(kubectl get svc emqx-listeners -o json | jq '.status.loadBalancer.ingress[0].ip')
 ```
+
+- Retrieve the dashboard IP address of the load balancer:
+```Shell
+lb_dashboard_ip=$(kubectl get svc emqx-dashboard -o json | jq '.status.loadBalancer.ingress[0].ip')
+```
+
+- connect, publish, and subscribe using MQTTX CLI
```
Comment (LenaLenaPan) [inline docs/en_US/deployment/on-gcp-gke.md:129]:
```suggestion
- Connect, publish, and subscribe using MQTTX CLI
```

File: docs/en_US/deployment/on-gcp-gke.md [hunk 17]
```diff
@@ -1,60 +1,172 @@
...
+
+- Access the EMQX dashboard
+```Shell
+http://${lb_ip}:18083
+```
+
+:::
+::::
+
+## Handing LoadBalancer TLS offloading
```
Comment (LenaLenaPan) [inline docs/en_US/deployment/on-gcp-gke.md:165]:
```suggestion
## Use LoadBalancer for TLS Offloading
```

File: docs/en_US/deployment/on-gcp-gke.md [hunk 18]
```diff
@@ -1,60 +1,172 @@
...
+
+Since Google LoadBalancer doesn't support TCP certificates, please check [discussion](https://github.com/emqx/emqx-operator/discussions/312) to address TCP certificate offloading issues.
+
+
+## Conclusion
```
Comment (LenaLenaPan) [inline docs/en_US/deployment/on-gcp-gke.md:170]: the conclusion part can be removed.

File: docs/en_US/deployment/on-gcp-gke.md [hunk 19]
```diff
@@ -1,60 +1,170 @@
...
 
-For details: [kubeconfig](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl)
 
-## Configure Cert Manager
+## Deploy EMQX on GKE
 
-To install `cert-manager` read the official docs:
+### Deploy EMQX Operator
+
+For the detailed steps on installing `cert-manager`, see the official documentation [Quick Start](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/getting-started/getting-started.md)
```
Comment (LenaLenaPan) [inline docs/en_US/deployment/on-gcp-gke.md]:
```suggestion
Install `cert-manager`. For the detailed steps on installing `cert-manager`, see the official documentation [Quick Start](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/getting-started/getting-started.md)
```

File: docs/en_US/deployment/on-gcp-gke.md [hunk 20]
```diff
@@ -1,60 +1,170 @@
...
 
-For details: [kubeconfig](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl)
 
-## Configure Cert Manager
+## Deploy EMQX on GKE
 
-To install `cert-manager` read the official docs:
+### Deploy EMQX Operator
+
+To install EMQX Operator, please check [Quick Start](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/getting-started/getting-started.md)
```
Comment (LenaLenaPan) [inline docs/en_US/deployment/on-gcp-gke.md:17]:
```suggestion
Install EMQX Operator. For detailed steps, see [Quick Start](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/getting-started/getting-started.md)
```

**Review by Rory-Z — APPROVED** [review]

**Pull request merged**